presto-main/src/main/java/com/facebook/presto/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -168,6 +170,14 @@ else if (!splitWaitingForAnyNode) {
         return new SplitPlacementResult(blocked, assignment);
     }
 
+    private ResettableRandomizedIterator<InternalNode> getRandomCandidates(int limit, NodeMap nodeMap)
+    {
+        if (randomCandidates.get() == null) {
+            randomCandidates.compareAndSet(null, new ResettableRandomizedIterator<>(selectNodes(limit, randomizedNodes(nodeMap, includeCoordinator, ImmutableSet.of()))));
```

Comment:
It's not intuitive that this returns the same list if it is already constructed regardless of the parameters. Is this stuff all done exactly once? If it's done multiple times why is it guaranteed parameters like limit haven't changed?

If this needs to be the case then should it error if the parameters have changed?